lib/ui/painting/codec.cc

```diff
@@ -377,15 +405,13 @@ sk_sp<SkImage> MultiFrameCodec::GetNextFrameImage(
     options.fFrameIndex = nextFrameIndex_;
     const int requiredFrame = frameInfos_[nextFrameIndex_].fRequiredFrame;
     if (requiredFrame != SkCodec::kNone) {
-      if (requiredFrame < 0 ||
-          static_cast<size_t>(requiredFrame) >= frameBitmaps_.size()) {
+      if (frameBitmaps_[requiredFrame]->bitmap_ == nullptr) {
         FML_LOG(ERROR) << "Frame " << nextFrameIndex_ << " depends on frame "
-                       << requiredFrame << " which out of range (0,"
-                       << frameBitmaps_.size() << ").";
+                       << requiredFrame << " which has not been cached.";
         return NULL;
       }
-      SkBitmap& requiredBitmap = frameBitmaps_[requiredFrame];
-      // For simplicity, do not try to cache old frames
+
+      SkBitmap requiredBitmap = *frameBitmaps_[requiredFrame]->bitmap_;
```

Comment:
nit: consider extracting a local ptr to `frameBitmaps_[requiredFrame]->bitmap_` from the condition in line 408, then using that here.